homeassistant/components/unifi/config_flow.py

```diff
@@ -118,13 +118,12 @@ async def async_step_user(
 
             else:
                 if (
-                    self.reauth_config_entry
-                    and self.reauth_config_entry.unique_id is not None
-                    and self.reauth_config_entry.unique_id in self.sites
+                    self.source == SOURCE_REAUTH
+                    and (reauth_unique_id := self._get_reauth_entry().unique_id)
+                    is not None
```

Comment:
```suggestion
                    and ((reauth_unique_id := self._get_reauth_entry().unique_id)
                    is not None)
```
If inline ifs span more than one line, rather add parenthesis so it will be easier to check